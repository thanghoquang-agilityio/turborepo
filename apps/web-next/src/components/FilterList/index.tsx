'use client'

import { memo } from 'react'

// Component
import { FilterList as SharedFilterList } from '@repo/next-ui'

import { ArrowDownIcon, ArrowUpIcon } from '@/icons'
import { useDebounce } from '@/hooks'
// Types
import type { FilterItem } from '@/types'

interface FilterListProps {
  title: string
  items: FilterItem[]
  isOpen: boolean
  defaultValue?: string[]
  additionalClass?: string
  handleReplaceURL?: (params: URLSearchParams) => void
}

const FilterList = memo(
  (props: FilterListProps) => {
    return (
      <SharedFilterList
        {...props}
        ArrowDownIcon={ArrowDownIcon}
        ArrowUpIcon={ArrowUpIcon}
        useDebounce={useDebounce}
      />
    )
  }
)

FilterList.displayName = 'FilterList'
export default FilterList
