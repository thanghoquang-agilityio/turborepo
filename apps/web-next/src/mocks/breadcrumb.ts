export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
