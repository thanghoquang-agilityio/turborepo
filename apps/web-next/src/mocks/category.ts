export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
