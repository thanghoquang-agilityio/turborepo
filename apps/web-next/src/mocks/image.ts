import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

export const MOCK_IMAGE = {
  data: {
    id: '24',
    attributes: {
      name: 'default',
      url: SRC_IMAGE_NOT_AVAILABLE,
    },
  },
}

export const MOCK_LIST_IMAGE_RESPONSE = [
  {
    name: 'default1',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default2',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default3',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default4',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
]
