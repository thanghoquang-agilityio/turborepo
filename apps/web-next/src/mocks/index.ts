export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
