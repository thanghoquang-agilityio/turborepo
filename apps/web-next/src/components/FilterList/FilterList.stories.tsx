import type { Meta, StoryObj } from '@storybook/react'

import FilterList from '@/components/FilterList'
// Mocks
import { MOCK_CATEGORY_LIST } from '@/mocks'

const meta = {
  title: 'Components/FilterList',
  tags: ['autodocs'],
  component: FilterList,
  parameters: {
    layout: 'centered',
    nextjs: {
      appDirectory: true,
    },
  },
} satisfies Meta<typeof FilterList>

export default meta
type Story = StoryObj<typeof FilterList>

export const Default: Story = {
  decorators: [
    (Story) => (
      <div className='mt-5 w-[304px] rounded-medium border-medium border-background-300 p-4'>
        <Story />
      </div>
    ),
  ],

  args: {
    title: 'Category',
    isOpen: true,
    items: MOCK_CATEGORY_LIST,
  },
}
