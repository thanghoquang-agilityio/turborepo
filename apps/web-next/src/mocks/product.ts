import { ProductVariantModel } from '@/types'

import { MOCK_IMAGE } from './image'

// New flattened product variants for testing
export const MOCK_PRODUCT_VARIANTS: ProductVariantModel[] = [
  {
    id: 1,
    color: 'slate blue',
    size: 'small,medium,large,extra large,XXL',
    stock: '20,20,20,20,20',
    tax: 5,
  },
  {
    id: 2,
    color: 'blue',
    size: 'small,medium,large,extra large,XXL',
    stock: '20,20,20,20,20',
    tax: 5,
  },
]

export const MOCK_PRODUCT = {
  id: 1,
  documentId: 'mock-product-1',
  image: MOCK_IMAGE,
  name: 'TDX Sinkers',
  variants: 5,
  price: 675.0,
  star: 5,
  comments: 121,
  description: '',
  discount: 0,
  likes: 11,
  brand: 'MOCK BRAND',
  productVariants: MOCK_PRODUCT_VARIANTS,
}

export const MOCK_PRODUCTS_RESPONSE = [
  {
    id: 1,
    documentId: 'mock-product-1',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 2,
    documentId: 'mock-product-2',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 3,
    documentId: 'mock-product-3',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 4,
    documentId: 'mock-product-4',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: MOCK_PRODUCT_VARIANTS,
    image: MOCK_IMAGE,
  },
]
