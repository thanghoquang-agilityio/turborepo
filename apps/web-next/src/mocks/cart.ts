import { CartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  documentId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
