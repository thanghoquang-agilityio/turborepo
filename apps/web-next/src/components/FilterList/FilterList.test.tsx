import { fireEvent, render, waitFor } from '@testing-library/react'

import { MOCK_CATEGORY_LIST } from '@/mocks'

import FilterList from '.'

describe('FilterList Component', () => {
  const renderComponent = (props = {}) => {
    return render(
      <FilterList
        title='Category'
        isOpen={true}
        items={MOCK_CATEGORY_LIST}
        {...props}
      />
    )
  }

  it('renders with default props', () => {
    const { container, getByTestId } = renderComponent()
    const toggleButton = getByTestId('btn-toggle-filter')

    fireEvent.click(toggleButton)
    expect(container).toMatchSnapshot()
  })

  it('handles selection change', () => {
    const { getByTestId } = renderComponent({ isOpen: true })
    const checkbox = getByTestId('checkbox-item-2')
    fireEvent.click(checkbox)

    waitFor(() => {
      expect(checkbox).toBeChecked()
    })
  })
})
