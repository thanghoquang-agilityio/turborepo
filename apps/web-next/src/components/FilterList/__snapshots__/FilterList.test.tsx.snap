// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList Component renders with default props 1`] = `
<div>
  <div
    class="flex max-w-[270px] flex-col items-center gap-[30px] "
  >
    <button
      aria-label="Button"
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none text-default-foreground data-[hover=true]:opacity-hover w-full bg-background-100"
      data-focus="true"
      data-focus-visible="true"
      data-testid="btn-toggle-filter"
      type="button"
    >
      <div
        class="flex h-6 w-full justify-between"
      >
        <p
          class="  no-underline text-ellipsis whitespace-nowrap overflow-hidden text-xl font-medium leading-5 text-secondary-200"
        >
          Category
        </p>
        <div
          class="flex"
        >
          <svg
            class="w-3.5 mb-2 mr-0.5"
            fill="none"
            viewBox="0 0 12 7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.375 5.3125L7 9.6875L2.625 5.3125"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
      </div>
      <span
        class="nextui-ripple"
        style="position: absolute; background-color: currentColor; border-radius: 100%; transform-origin: center; pointer-events: none; overflow: hidden; inset: 0; z-index: 0; top: 0px; left: 0px; width: 0px; height: 0px; transform: scale(0); opacity: 0.35; will-change: transform,opacity;"
      />
    </button>
  </div>
</div>
`;
